framework/ui/src/main/java/com/google/android/catalog/framework/ui/CatalogActivity.kt

```diff
@@ -75,12 +80,15 @@ open class CatalogActivity : FragmentActivity() {
                     modifier = Modifier.fillMaxSize(),
                     color = MaterialTheme.colorScheme.background
                 ) {
-                    CatalogNavigation(
-                        startDestination = startDestination,
-                        samples = catalogSamples,
-                        settings = settings,
-                        fragmentManager = supportFragmentManager,
-                    )
+                    val sizeClass = calculateWindowSizeClass(this)
+                    CompositionLocalProvider(LocalWindowSize provides sizeClass) {
```

Comment:
Generally we've been avoiding a composition local for the window size class, in favor of explicitly passing down the value wherever needed, and turning the size class into a specific behavior where needed.

The advantage there is more  clear `@Preview` support and testability, although the explicitness is more verbose.

Reply:
My first iteration was like that but I feel that it's needed in many screens and some standalone components benefits of being able to just understand the format from the composition. 

It don't fully see how it reduces Preview or testing support since in both cases you would need to anyways provide the value, either as parameter or as composition local.